azure-spring-boot/pom.xml

```diff
@@ -139,6 +139,11 @@
             <artifactId>adal4j</artifactId>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>com.microsoft.azure</groupId>
+            <artifactId>azure-client-authentication</artifactId>
+            <version>1.6.1</version>
```

Comment:
@jalanubha Thanks for the pull request, could you manage version in [BOM file](https://github.com/Microsoft/azure-spring-boot/blob/master/azure-spring-boot-bom/pom.xml) and write dependency without version here? 

If `azure-client-authentication` may not  be required by all auto-configuration classes, please add `<optional>true</optional>`.